OpenRA.Mods.Common/Traits/SupportPowers/GrantUpgradePower.cs

```diff
@@ -26,7 +26,7 @@ class GrantUpgradePowerInfo : SupportPowerInfo
 		[Desc("Duration of the upgrade (in ticks). Set to 0 for a permanent upgrade.")]
 		public readonly int Duration = 0;
 
-		[Desc("Cells")]
+		[Desc("Cells - affects whole cells only")]
 		public readonly int Range = 1;
```

Comment:
This is no different to any other code that has to query via cells.  You round the range up to the next full cell, find the candidate cells, query what you want, and then filter the results based on the actual (world) distances.